Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -190,6 +169,64 @@ public static string GetLanguageVersion(string targetFramework)
             return Unknown();
         }
 
+        // Knowledge on unique machine Ids for different OSes obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = System.Environment.OSVersion.Platform;
+            Match match = null;
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
+                {
+                    var machineIdToParse = RunProcess("reg", "QUERY HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography -v MachineGuid");
+                    var regex = new Regex("^\\s+MachineGuid\\s+\\w+\\s+((\\w+-?)+)", RegexOptions.Multiline);
+                    match = regex.Match(machineIdToParse);
+                }
+                else if (currentOs == PlatformID.MacOSX)
+                {
+                    var machineIdToParse = RunProcess("ioreg", "-rd1 -c IOPlatformExpertDevice");
+                    var regex = new Regex("^.*IOPlatformUUID\\\"\\s=\\s\\\"(.+)\\\"", RegexOptions.Multiline);
+                    match = regex.Match(machineIdToParse);
+                }
+                else if (currentOs == PlatformID.Unix)
+                {
+                    try
+                    {
+                        id = File.ReadAllText("/etc/machine-id");
+                    }
+                    catch
+                    {
+                        id = string.Empty;
+                    }
+                }
+
+                if (match?.Groups.Count > 1)
+                {
+                    id = match.Groups[1].Value;
+                }
+
+                if (id.Length == 0)
+                {
+                    return Unknown();
+                }
+                else
+                {
+                    var salt = new byte[] { 82, 101, 97, 108, 109, 32, 105, 115, 32, 103, 114, 101, 97, 116 };
```

Comment:
Did you get a reply from @nielsenko about why we're salting this? Looking at `node-machine-id` which is used by the js SDK, they're not salting the id.

Reply:
Yes I did. As expected it's just some low-effort extra precautions to avoid that the same machine can be easily identified across multiple products (and their metrics)

Reply:
Yes, it is just to avoid other unrelated projects to use the same id as us, if they use `node-machine-id`, which I believe is fairly common.

At the time I didn't actually try to have the same id as any other SDK, as it seemed very random what was used.

Reply:
Cool, can we add a comment that captures this. For the salt, we probably want to use the same one across our products to make sure we're identifying cases where the same developer uses multiple SDKs. While we believe this is pretty rare right now, we can't really confirm it and aligning on the hashed id would allow us to do so.

Reply:
Right, but the scope doc is fairly unrelated to the code in the repo. I doubt someone looking at the code in 2-3 years will think of opening the scope doc to understand why we're using a hardcoded value for the salt.